test/shared/src/main/scala/zio/test/Spec.scala

```diff
@@ -227,34 +227,36 @@ final case class Spec[-R, +E, +L, +T](caseValue: SpecCase[R, E, L, T, Spec[R, E,
     }
 
   /**
-   * Uses the specified `Managed` to provide each test in this spec with its
+   * Uses the specified `ZManaged` to provide each test in this spec with its
    * required environment.
    */
-  final def provideManaged[E1 >: E](managed: Managed[E1, R])(implicit ev: NeedsEnv[R]): Spec[Any, E1, L, T] =
-    transform[Any, E1, L, T] {
-      case SuiteCase(label, specs, exec) => SuiteCase(label, specs.provideManaged(managed), exec)
-      case TestCase(label, test)         => TestCase(label, test.provideManaged(managed))
+  final def provideManaged[R0, E1 >: E](managed: ZManaged[R0, E1, R])(implicit ev: NeedsEnv[R]): Spec[R0, E1, L, T] =
+    transform[R0, E1, L, T] {
+      case SuiteCase(label, specs, exec) => SuiteCase(label, specs.provideSomeManaged(managed), exec)
+      case TestCase(label, test)         => TestCase(label, test.provideSomeManaged(managed))
     }
 
   /**
-   * Uses the specified `Managed` once to provide all tests in this spec with
+   * Uses the specified `ZManaged` once to provide all tests in this spec with
    * a shared version of their required environment. This is useful when the
    * act of creating the environment is expensive and should only be performed
    * once.
    */
-  final def provideManagedShared[E1 >: E](managed: Managed[E1, R])(implicit ev: NeedsEnv[R]): Spec[Any, E1, L, T] = {
-    def loop(r: R)(spec: Spec[R, E, L, T]): ZIO[Any, E, Spec[Any, E, L, T]] =
+  final def provideManagedShared[R0, E1 >: E](
```

Comment:
I'd probably call this `provideSomeManagedShared` even though it's getting horrendously long (just for consistency).

The other way to approach "sharing" would be a combinator, e.g.:

```scala
shared {
  ...
}
```

Reply:
Okay. Agree about the value of consistency.